department-of-veterans-affairs/abd-vro
# less info for api argument validation

## What was the problem?


MethodArgumentNotValidException, which is raised when there are rest end-point validation, included stack information. This caused concern in WASA scans.

Associated tickets or Slack threads:

- [MCP-1992](https://amida.atlassian.net/browse/MCP-1992)

## How does this fix it?[^1]


Handles MethodArgumentNotValidException specifically and only  field name and reason of error for the field is put into the error message. Also it now returns 400 (Bad Request) for these errors instead of 500.

## How to test this PR
- Open swagger and enter erroneous inputs. Check there is no stack information.
- Step 2[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
